protocols/kad/src/behaviour.rs

```diff
@@ -680,37 +685,50 @@ where
     ///
     /// The result of this operation is delivered in a
     /// [`KademliaEvent::OutboundQueryCompleted{QueryResult::GetRecord}`].
-    pub fn get_record(&mut self, key: record::Key, quorum: Quorum) -> QueryId {
-        let quorum = quorum.eval(self.queries.config().replication_factor);
-        let mut records = Vec::with_capacity(quorum.get());
-
-        if let Some(record) = self.store.get(&key) {
+    pub fn get_record(&mut self, key: record::Key) -> QueryId {
+        let record = if let Some(record) = self.store.get(&key) {
             if record.is_expired(Instant::now()) {
-                self.store.remove(&key)
+                self.store.remove(&key);
+                None
             } else {
-                records.push(PeerRecord {
+                Some(PeerRecord {
                     peer: None,
                     record: record.into_owned(),
-                });
+                })
             }
-        }
+        } else {
+            None
+        };
```

Comment:
```suggestion
        }.flatten();
```

Should work as well, right?

Reply:
doesn't compile for me

Reply:
Would require a side-effect within a `Option::map`, which in my eyes is intuitive. Sorry for the noise.